content/2021/2021-07/2021-07-26-pulsar-weekly.md

```diff
@@ -0,0 +1,119 @@
+
+- [Broker] PIP-82: Update the resource-group stats.
+
+    https://github.com/apache/pulsar/pull/11331 ([kaushik-develop](https://github.com/kaushik-develop))
+
+- [C++ Client] Support the Protobuf native schema.
+
+    https://github.com/apache/pulsar/pull/11388 ([BewareMyPower](https://github.com/BewareMyPower))
+
+### Notable Feature
+
+- [Broker] Support setting multiples roles for authorization.
+
+    https://github.com/apache/pulsar/pull/11341 ([RobertIndie](https://github.com/RobertIndie))
+
+- [Python Client] Support the complex schema.
+
+    https://github.com/apache/pulsar/pull/11400 ([gaoran10](https://github.com/gaoran10))
+
+- [Pulsar IO] Allow Pulsar IO sink connectors to use the native AVRO and JSON formats.
+
+    https://github.com/apache/pulsar/pull/11322 ([eolivelli](https://github.com/eolivelli))
+
+- [Monitor] Add the `storageLogicalSize` metric to the `TopicStats` and `NamespaceStats`.
+
+    https://github.com/apache/pulsar/pull/11430 ([Technoboy-](https://github.com/Technoboy-))
+
+- [Broker] Close the replicator and replication client when deleting a Pulsar cluster.
+
+    https://github.com/apache/pulsar/pull/11342 ([codelipenghui](https://github.com/codelipenghui))
+
+### Notable Bug Fix
+
+- [CLI] Fix the issue in the pulsar-admin CLI tool when validating the batch source configuration.
+
+    https://github.com/apache/pulsar/pull/11378 ([jerrypeng](https://github.com/jerrypeng))
+
+- [Admin] Fix the issue that only subscriptions created for `Partition-0` are listed when partition-specific subscriptions are created.
+
+    https://github.com/apache/pulsar/pull/11355 ([Technoboy-](https://github.com/Technoboy-))
+
+- [Function] Reduce the probability of cache inconsistencies.
+
+    https://github.com/apache/pulsar/pull/11423 ([315157973](https://github.com/315157973))
+
+- [Transaction] Fix the direct memory leak related to the commit and abort markers.
+
+    https://github.com/apache/pulsar/pull/11407 ([lhotari](https://github.com/lhotari))
+
+- [Client] Avoid waiting infinitely when closing a consumer.
+
+    https://github.com/apache/pulsar/pull/11347 ([Shoothzj](https://github.com/Shoothzj))
+
+- [Broker] Delete the topic-level policies after deleting a topic.
+
+    https://github.com/apache/pulsar/pull/11316 ([horizonzy](https://github.com/horizonzy))
+
+- [Python Function] Fix the issue that the `tls_validate_hostname` is not supported in the Python Functions runtime.
+
+    https://github.com/apache/pulsar/pull/11087 ([freeznet](https://github.com/freeznet))
+
+- [C++ Client] Use the same regex code at `ZTSClient`.
+
+    https://github.com/apache/pulsar/pull/11323 ([equanz](https://github.com/equanz))
+
+- [Broker] Fix the issue that the broker leaks direct memory when using replicated subscriptions and Key_Shared consumers.
+
+    https://github.com/apache/pulsar/pull/11396 ([lhotari](https://github.com/lhotari))
+
+- [Broker] Fix the NPE that occurs when the builtin connectors do not exist.
```

Comment:
Fix the issue that NPE o

Reply:
NPE refers to Null Pointer Exception. Therefore, in some extend, an exception equal to an error. So, i don't think we need to use the `Fix the issue that NEP...` expression.